src/app/plugins/pubsub.plugin.ts

```diff
@@ -28,60 +32,51 @@ export class PubSubPlugin extends Plugin {
         const subs = response.data;
 
         this._subs = subs.map((subData: { name: string }) => {
-          return subData.name
-            .toLowerCase()
-            .split('-')
-            .join(' ');
+          return subData.name.toLowerCase();
         });
-        this._generateEmbedSubs();
       })
       .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    if (args === undefined || args.length === 0) {
-      args = [''];
-    }
+    const subIn = this._parseCommand(args || []);
 
-    if (args[0].includes('type') || args[0].includes('list')) {
+    if (subIn.includes('type') || subIn.includes('list')) {
       //Simply return the list of supported subs
+      this._generateEmbedSubs();
       message.reply(this._embedSubs);
       return;
     }
 
-    const subIn = this._parseCommand(args);
+    console.log(`${this._subs[0] === subIn}`);
 
     // checks if their sub was a sub, then if that sub is recognised
-    const subEntry = this._subs.find((sub: string) => sub === subIn) || '';
+    const subType = this._subs.find((sub: string) => sub === subIn) || '';
 
-    if (subIn == 'reboot') {
-      this.updateList();
-      return;
-    } else if (subEntry == '' && subIn != 'random' && subIn != '') {
+    if (subType === '' && subIn !== 'random' && subIn !== '') {
       message.reply('Sub not found.  Use "list" or "type" to get a list of possible subs');
       return;
     }
 
-    this.container.loggerService.debug(subEntry);
+    this.container.loggerService.debug(subType);
 
     //recieves the according info and posts, or derps
     await this.container.httpService
-      .get(`${this._API_URL}subs/?name=${subEntry.split(' ').join('-')}`)
+      .get(`${this._API_URL}subs/?name=${subType}`)
       .then((response: IHttpResponse) => {
-        const subData = response.data[0];
-        const saleInfo = subData.status
-          ? `a dicounted price of ${subData.price} until ${subData.last_sale.split('-')[1]}`
-          : `${subData.price}, the last sale Date was ${subData.last_sale}`;
+        const [subData] = response.data;
+        const parsedNamed = subData.sub_name.split('-').join(' ');
+
+        const saleInfo =
+          subData.status.toLowerCase() === 'true' //not sure why I have to do it this way but I do
```

Comment:
why not status.toBoolean or something?

Reply:
would have to make function for it, which I don't think is worth it.  As that just return it's truthy value.